cfgen/cfgen

```diff
@@ -1296,20 +1330,23 @@ class ConfPool(ToDhall):
                 parity_units = len(cluster_encls(m0conf)) - 1
             else:
                 assert t is PoolT.md
-                data_units = 1  # len(drives)
+                data_units = 1
                 parity_units = len(cluster_encls(m0conf)) - 1
-            pd_attrs = PDClustAttrs0(
-                data_units=data_units, parity_units=parity_units,
-                spare_units=0)
+            pd_attrs = PDClustAttrs0(data_units=data_units,
+                                     parity_units=parity_units,
+                                     spare_units=0)
         d = pool_desc.get('allowed_failures')
-        if t in (PoolT.sns, PoolT.dix) and d:
-            tolerance = Failures(d['site'], d['rack'], d['encl'], d['ctrl'],
-                                 d['disk'])
+        if t in (PoolT.sns, PoolT.dix):
+            if d:
+                tolerance = Failures(d['site'], d['rack'], d['encl'],
+                                     d['ctrl'], d['disk'])
+            else:
```

Comment:
I am wondering if the else condition at line 1358 which uses "gen_allowances()" is still applicable?

Reply:
Yeah, can you the generated tolerance vector directly. Fixed.